### DISCUSSION TITLE
What are the implications of putting the Plausible snippet in the <body>?

### DISCUSSION THREAD

**Discussion by firasm** [plausible/analytics#2117] [Anything Else]
Hello,

For a variety of reasons*, I would like to know what (if any) are the negative effects are of adding the snippet to the <body> of my HTML page, rather than the <head>?

I'm using JupyterBook and currently they don't yet have support for injecting html code into the `<head>`, rather only `.css` and `.js` files. 

> If you’d like to include custom CSS rules or JavaScript scripts in your book, add them to a folder called _static in your book’s folder. Any files that end in .css or .js in this folder will automatically be copied into your built book HTML and linked in the header of each page. ([from these docs](https://jupyterbook.org/en/stable/advanced/html.html?highlight=javascript#custom-assets))

My tracking seems to be working, and I am getting hits to my site, but I wonder if there are any negative consequences of the snippet NOT being in the head? I suspect I will lose some tracking info for people who don't let the whole site load, but my site loads in milliseconds, so maybe that's not such a big loss.

Thanks for making such a great product btw! SO glad I switched away from Google Analytics.

* This is mostly because I don't want to write a script to inject the snippet into the <head> when my static site generator (JupyterBook) currently only supports injecting html in the footer. I [have filed an issue requesting this feature](https://github.com/executablebooks/jupyter-book/issues/1811).

**Comment by RobertJoonas** [comment]
Hi @firasm!

As browsers start reading HTML from top to bottom, having our script in the `<head>` section is good, because it initiates the analytics event request as soon as the page is opened, and runs it asynchronously while rendering the page. In contrast, if you have it in the `<body>` section, the browser might have to wait until some images or other site content is loaded, before it can initiate the Plausible request.

In a bit of a hacky way, I believe you can still dynamically create the snippet in the <head> section on your site as well. From the docs you shared, it says that you can load custom `.js` files. So I think you could do something like this:

1) Create a JavaScript file and call it `add-plausible-snippet.js` for example
2) Paste the following code in that file:
```javascript
var script = document.createElement('script');
script.defer = true;
script.src = "https://plausible.io/js/script.js";
script.dataset.domain = "YOUR-DATA-DOMAIN";

// optional if using proxy
// script.dataset.api = 'https://yourproxy.com/api/event';

document.getElementsByTagName('head')[0].appendChild(script);
``` 
3) Modify the snippet attributes as you see fit (make sure to change "YOUR-DATA-DOMAIN" to your actual Plausible site name)
4) Save the file and move it to the `_static` folder as was instructed in the docs you shared

**Reply by firasm** [reply]
Clever ! That was a great idea!

I just implemented and it worked like a charm 👌. Thank you also for the explanation, that makes sense!

**Reply by firasm** [reply]
I've submitted a PR to add this trick to the [documentation](https://github.com/executablebooks/jupyter-book/pull/1814) - thanks again! 🙏🏽
